sway-core/src/ir_generation/const_eval.rs

```diff
@@ -20,14 +22,66 @@ use sway_types::{ident::Ident, span::Spanned};
 
 use std::collections::HashMap;
 
+pub struct LookupEnv<'a> {
+    pub context: &'a mut Context,
+    pub module: Module,
+    pub module_ns: Option<&'a namespace::Module>,
+    pub public_only: bool,
+    pub lookup: fn(&mut LookupEnv, &Ident) -> Result<Option<Value>, CompileError>,
+}
+
+pub fn compile_const_decl(
+    env: &mut LookupEnv,
+    name: &Ident,
+) -> Result<Option<Value>, CompileError> {
```

Comment:
What does returning `Ok(None)` signify?  It succeeded but there's no value?

Reply:
This function does two things
1. Provides a lookup.
2. Recursive entry to the actual const evaluator. 

In the case of a lookup not happening, it isn't the duty of this function to return an error, but just say `None`. In case the call to the actual evaluator fails, it passes on that error.